azure-quantum/azure/quantum/providers/ionq.py

```diff
@@ -0,0 +1,75 @@
+import io
+import json
+from typing import Any, Dict, Tuple
+
+from azure.quantum import Job
+from azure.quantum._client.models import JobDetails
+from azure.quantum.storage import upload_blob, ContainerClient
+from azure.quantum.workspace import Workspace
+
+IonQJson = Dict[str, Any]
+IONQ_INPUT_DATA_FORMAT = "ionq.circuit.v1"
+IONQ_OUTPUT_DATA_FORMAT = "ionq.quantum-results.v1"
+IONQ_PROVIDER_ID = "IonQ"
+DEFAULT_TIMEOUT = 100
+
+
+class IonQ:
+    def __init__(self, workspace: Workspace, target: str = "ionq.simulator"):
+        self.workspace = workspace
+        self.target = target
+    
+    def submit(
+        self,
+        circuit: IonQJson,
```

Comment:
The circuit parameter should be some opaque datastructure (byte[], stream) or some generic interface that just implements a `serialize` method such that it can be used regardless of how the actual format the provider expects. 
Even more important if we move this method to the `Job` class as I mentioned in my other comment. As a Job can have any type of payload.